src/openzeppelin/account.cairo

```diff
@@ -0,0 +1,233 @@
+use starknet::ContractAddress;
+use openzeppelin::utils::check_gas;
+
+const ERC165_ACCOUNT_ID: u32 = 0xa66bd575_u32;
+const ERC1271_VALIDATED: u32 = 0x1626ba7e_u32;
+
+const TRANSACTION_VERSION: felt252 = 1;
+// 2**128 + TRANSACTION_VERSION
+const QUERY_VERSION: felt252 = 340282366920938463463374607431768211457;
+
+#[derive(Serde, Drop)]
+struct Call {
+    to: ContractAddress,
+    selector: felt252,
+    calldata: Array<felt252>
+}
+
+#[abi]
+trait IAccount {
+    #[external]
+    fn __execute__(calls: Array<Call>) -> Array<Array<felt252>>;
+    #[external]
+    fn __validate__(calls: Array<Call>) -> felt252;
+    #[external]
+    fn __validate_declare__(class_hash: felt252) -> felt252;
+    #[external]
+    fn __validate_deploy__(
+        class_hash: felt252, contract_address_salt: felt252, _public_key: felt252
+    ) -> felt252;
+    #[external]
+    fn set_public_key(new_public_key: felt252);
+    #[view]
+    fn get_public_key() -> felt252;
+    #[view]
+    fn is_valid_signature(message: felt252, signature: Array<felt252>) -> u32;
+    #[view]
+    fn supports_interface(interface_id: u32) -> bool;
+}
+
+#[account_contract]
+mod Account {
+    use array::SpanTrait;
+    use array::ArrayTrait;
+    use box::BoxTrait;
+    use ecdsa::check_ecdsa_signature;
+    use serde::ArraySerde;
+    use starknet::get_tx_info;
+    use starknet::get_caller_address;
+    use starknet::get_contract_address;
+    use option::OptionTrait;
+    use zeroable::Zeroable;
+
+    use super::Call;
+    use super::ERC165_ACCOUNT_ID;
+    use super::ERC1271_VALIDATED;
+    use super::IAccount;
+    use super::TRANSACTION_VERSION;
+    use super::QUERY_VERSION;
+
+    use openzeppelin::introspection::erc165::ERC165;
+    use openzeppelin::utils::check_gas;
+    use openzeppelin::utils::span_to_array;
+
+    impl Account of IAccount {
+        fn __execute__(mut calls: Array<Call>) -> Array<Array<felt252>> {
+            // avoid calls from other contracts
+            // https://github.com/OpenZeppelin/cairo-contracts/issues/344
+            let sender = get_caller_address();
+            assert(sender.is_zero(), 'Account: invalid caller');
+
+            // check tx version
+            let tx_info = get_tx_info().unbox();
+            let version = tx_info.version;
+            if version != TRANSACTION_VERSION { // > operator not defined for felt252
+                assert(version == QUERY_VERSION, 'Account: invalid tx version');
```

Comment:
What's the purpose of the `operator not defined for felt252` comment?

Reply:
And can we test the assertion?

Reply:
> What's the purpose of the `operator not defined for felt252` comment?

`>` is the operator not defined. otherwise we would simply assert `version > TRANSACTION_VERSION` and skip the `QUERY_VERSION` check

Reply:
> And can we test the assertion?

+1